### DISCUSSION TITLE
DistributedDataParallel stuck on epoch 2

### DISCUSSION THREAD

**Discussion by 1zzt** [pyg-team/pytorch_geometric#8675] [Q&A]
`    train_sampler = DistributedSampler(train_dataset, num_replicas=world_size,
                                       rank=rank)
    train_loader = DataLoader(train_dataset, batch_size=batch_size, follow_batch=['x_1','x_2','x_3'],
                              sampler=train_sampler)
`
![image](https://github.com/pyg-team/pytorch_geometric/assets/22312961/06828575-0971-4957-981c-0986346f22d6)

**Comment by rusty1s** [comment]
Do you have a minimal example to reproduce? Is this a PyTorch or PyG issue?
